src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs

```diff
@@ -1837,35 +1831,110 @@ internal virtual void AggregateRenderPackages(IEnumerable<HelixRenderPackage> pa
-                        foreach (var range in sortedVerticesRange)
+                        RemoveMeshGeometryByRange(verticesRange, m);
+                    }
+
+                    //If we are using IRenderInstances data then we need to create a unique Geometry3D object for each instancable item and add instance transforms.  
+                    //If we any mesh geometry was not associated with an instance, remove the previously added mesh data from the render package so the remaining mesh can be added to the scene.
+                    if (rp.MeshVerticesRangesAssociatedWithInstancing.Any())
+                    {
+                        //For each range of mesh vertices add the mesh data and instances to the scene
+                        var meshVertexCountTotal = 0;
+                        var j = 0;
+                        foreach (var item in rp.MeshVerticesRangesAssociatedWithInstancing)
+                        {
+                            var range = item.Value;
+                            var index = range.Item1; //Start mesh vertex index
+                            var count = range.Item2 - range.Item1 + 1; //Count of mesh vertices
+                            var uniqueId = baseId + ":" + j + MeshKey;
+
+                            List<Matrix> instances;
+                            if (rp.instances.TryGetValue(item.Key, out instances))
+                            {
+                                AddMeshData(uniqueId, rp, index, count, drawDead, baseId, rp.Colors,
+                                    rp.ColorsStride, instances);
+                            }
+
+                            //Track cumulative total of mesh vertices added.
+                            meshVertexCountTotal += count;
+                            j++;
+                        }
+
+                        //If all the mesh regions had texture map data then we are done with mesh data.
+                        if (meshVertexCountTotal == m.Positions.Count)
                         {
-                            var i = range.Item1;
-                            var c = range.Item2 - range.Item1 + 1;
-                            m.Positions.RemoveRange(i,c);
-                            m.Colors.RemoveRange(i, c);
-                            m.Normals.RemoveRange(i, c);
-                            m.TextureCoordinates.RemoveRange(i, c);
+                            continue;
                         }
 
-                        //Regenerate Indices data
-                        var sequence = Enumerable.Range(0, m.Positions.Count);
-                        var newIndices = new IntCollection();
-                        newIndices.AddRange(sequence);
-                        m.Indices = newIndices;
+                        //Otherwise, clean up the remaining mesh geometry data in the render package to exclude the regions already generated.
+                        var verticesRange =
```

Comment:
Is it a single vertex range or multiple vertex ranges? If it's the latter, I believe this should be `vertexRanges`, correct?